feat(manager/fleet): implement Rancher Fleet manager (#16138)
renovatebot/renovate: lib/modules/manager/fleet/readme.md

> Allows to upgrade bundle definitions and GitRepo yaml manifest of Rancher Fleet.

Reviewer: ```suggestion
Can upgrade bundle definitions and GitRepo YAML manifests of Rancher Fleet.
```

> By default, only bundles with Helm references will be upgraded. To enable GitRepo updates you have to extend your [fileMatch](https://docs.renovatebot.com/configuration-options/#filematch) configuration

Reviewer: ```suggestion
By default, only bundles with Helm references will be upgraded.
To enable GitRepo updates you have to extend your [`fileMatch`](https://docs.renovatebot.com/configuration-options/#filematch) configuration.
```

Can upgrade bundle definitions and GitRepo YAML manifests of Rancher Fleet.

By default, only bundles with Helm references will be upgraded.
To enable GitRepo updates you have to extend your [`fileMatch`](https://docs.renovatebot.com/configuration-options/#filematch) configuration.

```json
{
  "fleet": {

>     "fileMatch": ["myGitRepoManifests\\.yaml"]

Reviewer: Is this `\\` correct?

Reviewer 2: yes, we need to double escape regex in strings
